openpype/hosts/maya/plugins/publish/extract_maya_scene_raw.py

```diff
@@ -1,13 +1,117 @@
 # -*- coding: utf-8 -*-
 """Extract data as Maya scene (raw)."""
 import os
+import contextlib
 
 from maya import cmds
 
 from openpype.hosts.maya.api.lib import maintained_selection
 from openpype.pipeline import AVALON_CONTAINER_ID, publish
 
 
+def offset_node(node, offset):
+    node_type = cmds.nodeType(node)
+    if node_type.startswith("animCurve"):
+        cmds.keyframe(node, edit=True, relative=True, timeChange=offset)
+
+    elif node_type == "imagePlane":
+        node_attr = node + ".frameOffset"
+        cmds.setAttr(node_attr, cmds.getAttr(node_attr) + (-offset))
+
+    elif node_type == "timeSliderBookmark":
+        for attr in ["timeRangeStart", "timeRangeStop"]:
+            node_attr = "{}.{}".format(node, attr)
+            cmds.setAttr(node_attr, cmds.getAttr(node_attr) + offset)
+
+
+@contextlib.contextmanager
+def offset_scene(offset):
+    # Exit early for easier stacking of context managers.
+    if offset is None:
+        yield
+        return
+
+    nodes = cmds.ls(type=("animCurve", "imagePlane", "timeSliderBookmark"))
+    changed_nodes = []
+    try:
+        for node in nodes:
+            offset_node(node, offset)
+            changed_nodes.append(node)
+        yield
+    finally:
+        for node in changed_nodes:
+            offset_node(node, -offset)
+
+
+@contextlib.contextmanager
+def maintain_timeline(frame_start, frame_end, handle_start, handle_end):
+    # Exit early for easier stacking of context managers.
+    if None in [frame_start, frame_end, handle_start, handle_end]:
+        yield
+        return
+
+    data = {
+        "minTime": None,
+        "maxTime": None,
+        "animationStartTime": None,
+        "animationEndTime": None
+    }
+    for key in data.keys():
+        kwargs = {key: True, "query": True}
+        data[key] = cmds.playbackOptions(**kwargs)
+
+    current_time = cmds.currentTime(query=True)
+    render_start_frame = cmds.getAttr("defaultRenderGlobals.startFrame")
+    render_end_frame = cmds.getAttr("defaultRenderGlobals.endFrame")
+
+    try:
+        animation_start_time = frame_start - handle_start
+        animation_end_time = frame_end + handle_start
+
+        cmds.playbackOptions(
+            minTime=frame_start,
+            maxTime=frame_end,
+            animationStartTime=animation_start_time,
+            animationEndTime=animation_end_time
+        )
+
+        cmds.setAttr(
+            "sceneConfigurationScriptNode.before",
+            "playbackOptions -min {} -max {} -ast {} -aet {}".format(
+                frame_start,
+                frame_end,
+                animation_start_time,
+                animation_end_time
+            ),
+            type="string"
+        )
```

Comment:
Why do we actually need to explicitly update this UI script? Doesn't Maya do that automatically on save or export so that whatever gets extracted has those values? This seems quite a low level access that I'm not sure we'd really need to do?

Reply:
Yeah, we need to update this UI script.